use-product-search.md

```diff
@@ -0,0 +1,304 @@
+---
+title: Using the Product Search
+excerpt: It is possible to find products by various criteria using the `ProductSearch` class, including name, slug, taxons, channels, price ranges, sorting, and custom attributes.
+date: 2025-02-17
+tags: [models, product]
+---
+
+The `ProductSearch` class provides a powerful and flexible way to retrieve products based on various criteria. 
+
+It supports:
+
+> - Filtering by name, slug, taxon, price range, and property values
+> - Sorting options and pagination
+> - Custom attribute filtering
+
+For more details, check out the Vanilo.io [documentation](https://vanilo.io/docs/4.x/products).
+
+This guide walks you through setting up product listing, implementing filtering, and displaying individual product details.
+
+## Product Listing
+
+The product listing page retrieves and displays all available products.
+
+#### Controller:
+
+```php
+namespace App\Http\Controllers;
+
+use Vanilo\Foundation\Search\ProductSearch;
+
+class ProductController extends Controller
+{
+    private ProductSearch $productFinder;
+
+    public function __construct(ProductSearch $productFinder)
+    {
+        $this->productFinder = $productFinder;
+    }
+    
+    public function index()
+    {
+        return view('product.index', [
+            'products' => $this->productFinder->getResults(),
+        ]);
+    }
+}
+```
+
+#### Route:
+
+```php
+use App\Http\Controllers\ProductController;
+
+Route::get('/product-list', [ProductController::class, 'index'])->name('product.index');
+```
+
+#### View:
```

Comment:
Do not abuse headings: https://www.maine.edu/content-management/writing-style-guide/headings/

> ### What not to do with headings
>
>- **Do not use headings to make a paragraph of text stand out**. Instead, put a heading over the text to characterize it better.
> - Do not use bold instead of a heading. Both screen readers and search engines will not recognize this as a heading, and the benefits of having the heading are diminished. [Click here to read guidelines for how to use bold.](https://www.maine.edu/content-management/writing-style-guide/emphasis/)